src/game/behaviors/fish.inc.c

```diff
@@ -1,163 +1,286 @@
+    struct Animation **fishAnimation;
+    struct Object *fishObject;
     switch (o->oBehParams2ndByte) {
+        
+        // Blue fish with a quanitiy of twenty.
         case 0:
-            model = MODEL_FISH;            sp38 = 20;            sp30 = 1500.0f;            sp2C = blue_fish_seg3_anims_0301C2B0;
+            model = MODEL_FISH;
+            schoolQuantity = 20;
+            minDistToMario = 1500.0f;
+            fishAnimation = blue_fish_seg3_anims_0301C2B0;
             break;
+            
+        // Blue fish with a quanitiy of five.
         case 1:
-            model = MODEL_FISH;            sp38 = 5;            sp30 = 1500.0f;            sp2C = blue_fish_seg3_anims_0301C2B0;
+            model = MODEL_FISH;
+            schoolQuantity = 5;
+            minDistToMario = 1500.0f;
+            fishAnimation = blue_fish_seg3_anims_0301C2B0;
             break;
+            
+        // Cyan fish with a quanitiy of twenty.
         case 2:
-            model = MODEL_CYAN_FISH;            sp38 = 20;            sp30 = 1500.0f;            sp2C = cyan_fish_seg6_anims_0600E264;
+            model = MODEL_CYAN_FISH;
+            schoolQuantity = 20;
+            minDistToMario = 1500.0f;
+            fishAnimation = cyan_fish_seg6_anims_0600E264;
             break;
+            
+        // Cyan fish with a quanitiy of five.
         case 3:
-            model = MODEL_CYAN_FISH;            sp38 = 5;            sp30 = 1500.0f;            sp2C = cyan_fish_seg6_anims_0600E264;
+            model = MODEL_CYAN_FISH;
+            schoolQuantity = 5;
+            minDistToMario = 1500.0f;
+            fishAnimation = cyan_fish_seg6_anims_0600E264;
             break;
     }
-    if (o->oDistanceToMario < sp30 || gCurrLevelNum == LEVEL_SA) {
-        for (i = 0; i < sp38; i++) {
-            sp28 = spawn_object(o, model, bhvFishGroup2);
-            sp28->oBehParams2ndByte = o->oBehParams2ndByte;
-            func_8029EE20(sp28, sp2C, 0);
-            translate_object_xyz_random(sp28, 700.0f);
+    /**
+     * If Distance to Mario is lower than 1500 or the current level is Secret Aquarium,
+     * then spawn and animate the schoolQuantity of fish.
+     * Fish moves at random with a range of 700.0f.
+     */
+    if (o->oDistanceToMario < minDistToMario || gCurrLevelNum == LEVEL_SA) {
+        for (i = 0; i < schoolQuantity; i++) {
+            fishObject = spawn_object(o, model, bhvFishGroup2);
+            fishObject->oBehParams2ndByte = o->oBehParams2ndByte;
+            func_8029EE20(fishObject, fishAnimation, 0);
+            translate_object_xyz_random(fishObject, 700.0f);
         }
         o->oAction = 1;
     }
 }
 
-void ActionFish1(void) {
-    if (gCurrLevelNum != LEVEL_SA)
-        if (gMarioObject->oPosY - o->oPosY > 2000.0f)
+/**
+ * If the current level is not Secret Aquarium and the distance from
+ * mario minus o->oPosY is greater than 2000.0f then set action to two.
+ */
+void fish_action_chkDistance(void) {
```

Comment:
snake_case

i.e. fish_action_chk_distance

However, 2 things-
1) write out check
2) make either _action_ or _act_ consistent. _act_ is preferable imo